## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
!pip install statsmodels

   ---------------------------------------- 0.0/9.8 MB ? eta -:--:--
   -------------------- ------------------- 5.0/9.8 MB 33.7 MB/s eta 0:00:01
   ---------------------------------------- 9.8/9.8 MB 43.9 MB/s eta 0:00:00
   ---------------------------------------- 0.0/43.6 MB ? eta -:--:--
   ----------- ---------------------------- 12.6/43.6 MB 60.7 MB/s eta 0:00:01
   --------------------- ------------------ 23.3/43.6 MB 73.9 MB/s eta 0:00:01
   ----------------------------- ---------- 32.5/43.6 MB 52.9 MB/s eta 0:00:01
   ---------------------------------------  43.5/43.6 MB 53.3 MB/s eta 0:00:01
   ---------------------------------------- 43.6/43.6 MB 50.5 MB/s eta 0:00:00


In [2]:
import pandas as pd
import numpy as np
df = pd.read_csv('../data/data.csv')

data = df['data'].to_numpy()

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [3]:
data.shape

(100,)

In [5]:
data

array([17.49945292, 19.66239904,  7.18282343, 29.84162485,  9.23938649,
       14.94951581, 23.0172885 , 12.52827148, 13.65228669, 14.54595431,
       23.30872427, 33.75324808, 20.24923636, 13.29244733, 23.23434992,
       16.4230418 , 19.88521702, 27.05000732, 15.5127743 , 20.05415151,
       14.73135264, 19.06139498, 21.53942271, 14.06732571, 17.9670682 ,
       18.58289581, 16.17406993, 12.87432628, 11.47269664, 19.07902883,
       18.38565824, 33.38820073,  5.39139454, 20.67635903, 22.22266722,
       28.15780318, 23.01114324, 14.93471778, 20.00005857, 23.25411543,
       18.11895082, 24.62607043,  8.79145607, 30.387108  , 28.80606806,
       17.98593597, 23.66804468, 20.28782355, 15.02518827, 20.52626131,
       26.00219532, 17.71344489, 17.74598346, 19.55317542, 22.60097798,
       27.67027538, 16.19192417, 23.05037746, 21.29669604,  8.84832568,
       17.48410111, 19.20602661, 19.76257856, 21.9560206 ,  7.75806171,
       20.27753314, 15.93394654, 11.36336584, 23.14577858, 24.41

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [6]:
from statsmodels.stats.weightstats import ztest

In [12]:
# Нульова гіпотеза (H0): Середній чек дорівнює 20 доларам
# Альтернативна гіпотеза (H1): Середній чек більше 20 доларів

null_mean = 20
alpha = 0.05

ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')

In [10]:
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

In [11]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


p-value показує ймовірність отримати такі або ще більш екстремальні результати при умові, що нульова гіпотеза є правильною.
Якщо p-value <= alpha, ми відхиляємо нульову гіпотезу.   

У нашому випадку 0,84 > 0.05, тому у нас немає статистичних підстав відхилити нульову гіпотезу.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [13]:
df_2 = pd.read_csv('../data/data_2.csv')

data_2 = df_2['data'].to_numpy()

In [14]:
np.mean(data), np.mean(data_2)

(19.37755319670441, 19.37322276512519)

In [15]:
ztest(data, data_2, value=0)

(0.005246673456808504, 0.9958137794593356)

Статистика для двох вибірок z-критерію становить 0.005, а відповідне p-value дорівнює 0.995.

Оскільки p-value > 0.05, у нас немає доказів, щоб відкинути нульову гіпотезу, тобто середній чек в серпні місяці статистично не відрізняється від липня місяця. Порівнявши, середні значення через `np.mean`, спостерігаємо, що цей показник практично однаковий для обох місяців.